### PR TITLE
feat: Implement log cheatcodes

### PR DESCRIPTION
**Pull request by Jrigada**
# What :computer: 
* Added recordLogs and getRecordedLogs cheatcodes

# Evidence :camera:
image

<!-- # Notes :memo:
* As of right now we are not able to return a vector of logs. This would be the required functionality but we still can't decode the Log vector coming from the execution as return data.

### CODE REVIEW

**Comment by dutterbutter** [comment]
Please address conflicts @Jrigada

File: crates/era-cheatcodes/src/cheatcodes.rs [hunk 1]
```diff
@@ -296,13 +339,50 @@ impl CheatcodeTracer {
                 tracing::info!("👷 Returndata is {:?}", account_nonce);
                 self.return_data = Some(vec![account_nonce]);
             }
+            getRecordedLogs(getRecordedLogsCall {}) => {
+                tracing::info!("👷 Getting recorded logs");
+                let log_vector: Vec<Log> = self
```
Comment (nbaztec) [inline crates/era-cheatcodes/src/cheatcodes.rs:344]: We can simply call this `logs`. We should avoiding naming things after their types as it is easily inferred in strongly typed languages.

File: crates/era-cheatcodes/src/cheatcodes.rs [hunk 2]
```diff
@@ -296,13 +339,50 @@ impl CheatcodeTracer {
...
+                        topics: vec![log.topic.to_fixed_bytes().into()],
+                        data: log.data.to_fixed_bytes().into(),
+                        emitter: log.emitter.to_fixed_bytes().into(),
+                    })
+                    .collect_vec();
+                //keep only first element for testing purposes
+
+                let result = getRecordedLogsReturn { logs: log_vector };
+
+                println!("👷 Returndata is {:?}", result);
```
Comment (nbaztec) [inline crates/era-cheatcodes/src/cheatcodes.rs:357]: Maybe this was missed as a `info!`

File: crates/era-cheatcodes/src/cheatcodes.rs [hunk 3]
```diff
@@ -669,3 +750,10 @@ impl CheatcodeTracer {
         self.return_data = Some(data);
     }
 }
+fn transform_to_logs(events: Vec<EventMessage>, emitter: H160) -> Vec<LogEntry> {
+    events
+        .iter()
+        .filter(|event| event.address == zksync_types::EVENT_WRITER_ADDRESS)
+        .map(|event| LogEntry::new(u256_to_h256(event.key), u256_to_h256(event.value), emitter))
```
Comment (nbaztec) [inline crates/era-cheatcodes/src/cheatcodes.rs:757]: optional: there's a `filter_map` available for this usecase

**Review by nbaztec — APPROVED** [review]
small nits

**Pull request merged**